### PR TITLE
Use numeric_limits instead of magic number

### PR DESCRIPTION
**Pull request by alagner**

### CODE REVIEW

File: istring.cc [hunk 1]
```diff
@@ -54,7 +55,7 @@ int strncasecmp(const char *s1, const char *s2, uint32 length) {
 }
 
 int strcasecmp(const char *s1, const char *s2) {
-	return strncasecmp(s1, s2, 2147483647);
+	return strncasecmp(s1, s2, std::numeric_limits<int32_t>::max());
```
Comment (marzojr) [inline istring.cc:58]: Hm. I have no idea why it was just not using strcasecmp instead; could you please change to just remove the third parameter and include and use strcasecmp instead?

Comment (alagner) [reply istring.cc:58]: You mean to use the library function? My guess it's not been use before because it's POSIX, not standard library.

Comment (marzojr) [reply istring.cc:58]: Both strncasecmp and strcasecmp are POSIX functions (POSIX.1-2001, according to the man page), so no difference there.

Comment (alagner) [reply istring.cc:58]: Yeah, but see the whole file. `strncasecmp` is actually implemented some lines above my fix (`istring.cc:34`). The call refers to `Pentagram::strncasecmp`, not the global one. Unless Exult's implementation can be ifdefed-away and built only on Windows.

**Comment by marzojr** [comment]
I was taking a closer look at the file, and I will reimplement your solution while rewriting the functions. I just noticed that as-is, they are actually broken.

**Comment by marzojr** [comment]
For some reason I never got a notification email, so sorry for the delay.

Anyway: you are correct, I forgot we reimplemented those functions. Looking over, could I trouble you to do a couple changes? Specifically:
* modify `istring.h` to change `strncasecmp` to use `size_t` instead of `uint32`;
* do the same in `istring.cc`;
* change your `std::numeric_limits` change to also use `size_t`.

`size_t` is the correct type to use here, even though we probably will not be using strings that large anytime soon.

**Comment by alagner** [comment]
> For some reason I never got a notification email, so sorry for the delay.
> 
> Anyway: you are correct, I forgot we reimplemented those functions. Looking over, could I trouble you to do a couple changes? Specifically:
> 
> * modify `istring.h` to change `strncasecmp` to use `size_t` instead of `uint32`;
> * do the same in `istring.cc`;
> * change your `std::numeric_limits` change to also use `size_t`.
> 
> `size_t` is the correct type to use here, even though we probably will not be using strings that large anytime soon.

No problem with that. What about compiling those functions conditionally on Windows?

**Comment by marzojr** [comment]
Compiling them conditionally on Windows would run on the issue of several places using the full qualified Pentagram namespace to refer to it (e.g., on imagewin/imagewin.cc). It would require a bigger change. If you are up for it, it would be appreciated.

**Comment by alagner** [comment]
Came up with the idea of deducing the type, one place is nothing, but maybe it saves some maintenance in multiple places also. Will revert that in case it fails to build.

**Comment by marzojr** [comment]
Deducing the type there is overkill, and I would rather not; the POSIX version is specified as using size_t on that parameter, and that is unlikely to ever change.

**Comment by alagner** [comment]
Done. Unless you want me to squash that all into a single commit to increase readability.

**Comment by marzojr** [comment]
Please squash, yes.

**Pull request merged**